integration_tests/src/main/python/parquet_test.py

```diff
@@ -779,6 +780,38 @@ def test_spark_32639(std_input_path):
         lambda spark: spark.read.schema(schema_str).parquet(data_path),
         conf=original_parquet_file_reader_conf)
 
+@pytest.mark.skipif(not is_before_spark_320(), reason='Spark 3.1.x does not need special handling')
+def test_spark_40819_31x(std_input_path):
```

Comment:
Nit: Personally I'd rather see these tests named something more descriptive, e.g.: test_parquet_read_timestamp_nano_31x. 
 Otherwise as a reader I need to figure out what "40819" means, and even if I know that it is a Spark JIRA number, I have to lookup the Spark JIRA to understand what it's testing.